app/src/main/java/net/squanchy/service/firebase/FirebaseDbService.java

```diff
@@ -37,8 +38,10 @@
     private static final String TRACKS_NODE = "data/tracks";
     private static final String VENUE_INFO_NODE = "data/venue";
     private static final String TRACKS_BY_ID_NODE = "data/tracks/%1$s";
-    private static final String FAVORITES_NODE = "user/%1$s/";
-    private static final String FAVORITES_BY_ID_NODE = "user/%1$s/favorites/%2$s";
+    private static final String FAVORITES_NODE = "user/%1$s/favorites";
+    private static final String FAVORITES_BY_ID_NODE = "user/%1$s/favorites/map/%2$s";
```

Comment:
The optional pipeline in the getter for favourite s
(optionalFavorites -> optionalFavorites.or(FirebaseFavorites.empty())) 
stopped working because FirebaseFavourites was mapped on the entire sub-user space (even if it only mapped favourites) and there actually might be an hashmap there that is not called "favourites" (the mapping for the hashmap is not done by defining it in the reference path, rather by having it as a field in FirebaseFavourites). As an effect "data" is not null, but the map is.
So I divided the user space in 2: favourites and achievements, and each of them has its own map.
Would you suggest a name change for the field or maybe having a generic FirebaseUserdata object with both favourites and achievements?